### PR TITLE
Display correct values for the retention in the flags web gui.

### PR DESCRIPTION
**Pull request by krasi-georgiev**
fixes https://github.com/prometheus/prometheus/issues/5321

Signed-off-by: Krasi Georgiev <kgeorgie@redhat.com>

### CODE REVIEW

**Comment by codesome** [comment]
This goes in 2.8.0

**Comment by roidelapluie** [comment]
This will change the value in the flags. But it is confusing: if I add a maxbytes retention, the retention flag will change even if I do not change the retention flag.


I would propose to move that information to the Runtime and Build info page, by creating two metrics:

tsdb_retention_bytes
And
tsdb_retention_seconds

What do you think?

That could go in 2.9 ofc.

**Comment by krasi-georgiev** [comment]
> This will change the value in the flags. But it is confusing: if I add a maxbytes retention, the retention flag will change even if I do not change the retention flag.

the flag will change only when no time or size retention is set and the default of 15d is applied. The behaviour hasn't changed since 2.7

```
if cfg.tsdb.RetentionDuration == 0 && cfg.tsdb.MaxBytes == 0 {
```



```
I would propose to move that information to the Runtime and Build info page, by creating two metrics:

tsdb_retention_bytes
And
tsdb_retention_seconds

What do you think?
```

I don't see why would we move it under Runtime and Build. 
These are flags and looks logical to stay under the /flags web page.

**Comment by brian-brazil** [comment]
I'd expect flags to be what the user passed in, and runtime to show current status (i.e. what actually applies).

**Comment by krasi-georgiev** [comment]
> I'd expect flags to be what the user passed in, and runtime to show current status (i.e. what actually applies).

aaah yeah that is a good way to think about it. otherwise if you miss the logs it is weird to see 15d of retention when you didn't pass anything as a flag.

I like the idea and will look into it.

**Comment by roidelapluie** [comment]
Yes @bbrazil explained exactly what I had in mind.

**Comment by krasi-georgiev** [comment]
updated!

now the /flags page will show exactly what was passed where the /status will show what is the actual retention values.

Flags page
Screenshot_20190311_104324

Status Page
Screenshot_20190311_104254

File: web/ui/templates/status.html [hunk 1]
```diff
@@ -45,6 +45,14 @@ <h2 id="runtime">Runtime Information</h2>
...
           <td>{{.GOGC}}</td>
         </tr>
+        <tr>
+          <th>Storage Time Retention</th>
+          <td>{{.RetentionTime}}</td>
+        </tr>
+        <tr>
+          <th>Storage Size Retention</th>
+          <td>{{.RetentionSize}}</td>
+        </tr>
```
Comment (roidelapluie) [inline web/ui/templates/status.html]: Can we simplify this and have just one line?

(exepmle values that would be. printed: 
- 15d
- 15d or 10GB
- 19GB
)

Comment (krasi-georgiev) [reply web/ui/templates/status.html]: yep , not a bad idea. I updated the code.

Comment (krasi-georgiev) [reply web/ui/templates/status.html]: Screenshot_20190311_110834
Screenshot_20190311_110908

**Pull request merged**